### DISCUSSION TITLE
Devices Sync progress -  listening on different devices

### DISCUSSION THREAD

**Discussion by faridyou** [PaulWoitaschek/Voice#1326] [Ideas]
The idea is to be able to listen to a book on a device, then be able to continue listening on another device right where you left in the first device.
We can have this feature implemented in three modes, or phases:
- give the possibility to do this manually, like take a "status file",  and copy it over to the other device, use any file sync and share service to do so, like Google drive. This also assumes content and Mp3 is shared manually between the two devices. 
-  semi automatic,  where only Mp3 are shared manually 
- full authentication, where content and status are synced fully on the go

This would be a killer feature.

**Comment by ahmuro** [comment]
This would definitely be nice. Since a full auto sync would require a server side probably is not very feasible because of the complexity and mainly the cost. However status sync using personal cloud like Google drive would absolutely be nice to have.

**Comment by pacjo** [comment]
How about storing data in a directory (maybe in main storage or documents), that you could sync with something like syncthing?

**Comment by ahmuro** [comment]
> How about storing data in a directory (maybe in main storage or documents), that you could sync with something like syncthing?

This probably would be the most feasible option. Put the data in a file that's automatically and continually exported and then the user can set up their own sync service to wherever storage to sync that data file.

**Comment by Jesse-Bakker** [comment]
For me, the best solution for this would be for Voice to send a [broadcast] with the current audiobook progress data and to register a broadcast receiver for a similar intent to receive progress. This would allow third party implementations for sync protocols, installable as third party apps.

[broadcast]: https://developer.android.com/guide/components/broadcasts

**Comment by pho3nixf1re** [comment]
Moon+ Reader uses WebDav, Dropbox API, and Google Drive as options. Personally I use the WebDav to my own NextCloud server. That may be a good model for an app like this.

**Comment by LeonardoGentile** [comment]
Hey there any update on this?

**Comment by Penguinjumper** [comment]
Easiest way would be to support syncthing or nextcloud by just saving the bookmarks and progress in a small txt file in the individual audiobook's folder
